src/main/scala/mesosphere/mesos/TaskBuilder.scala

```diff
@@ -29,20 +29,31 @@ class TaskBuilder(app: AppDefinition,
 
   def buildIfMatches(offer: Offer): Option[(TaskInfo, Seq[Long])] = {
 
-    ResourceMatcher.matchResources(offer, app, taskTracker.get(app.id)) match {
-      case Some(ResourceMatch(cpu, mem, disk, ranges)) =>
-        build(offer, cpu, mem, disk, ranges)
+    val acceptedResourceRoles: Set[String] = app.acceptedResourceRoles.getOrElse(config.defaultAcceptedResourceRolesSet)
 
-      case _ =>
-        log.info(
-          s"No matching offer for ${app.id} (need cpus=${app.cpus}, mem=${app.mem}, " +
-            s"disk=${app.disk}, ports=${app.hostPorts()}) : " + offer
-        )
-        None
+    if (log.isDebugEnabled) {
+      log.debug(s"acceptedResourceRoles $acceptedResourceRoles")
     }
+
+    ResourceMatcher.matchResources(
+      offer, app, taskTracker.get(app.id),
+      acceptedResourceRoles = acceptedResourceRoles) match {
```

Comment:
because the type of acceptedResourceRoles is fairly generic (Set[String]) and I like to use named parameters to avoid confusion in these cases.